src/HealthGPS/mapping.cpp

```diff
@@ -8,27 +8,13 @@
 
 namespace hgps {
 
-MappingEntry::MappingEntry(std::string name, int level, core::Identifier entity_key,
-                           FactorRange range)
-    : name_{name}, name_key_{core::Identifier{name}}, level_{level},
-      entity_key_{std::move(entity_key)}, range_{range} {}
-
-MappingEntry::MappingEntry(std::string name, int level, core::Identifier entity_key)
-    : MappingEntry(name, level, entity_key, FactorRange{}) {}
-
-MappingEntry::MappingEntry(std::string name, int level)
-    : MappingEntry(name, level, core::Identifier::empty(), FactorRange{}) {}
+MappingEntry::MappingEntry(std::string name, int level, FactorRange range)
+    : name_{name}, name_key_{core::Identifier{name}}, level_{level}, range_{range} {}
```

Comment:
You're copying `name` twice, but you could do it with one move and one copy instead:

```suggestion
    : name_{std::move(name)}, name_key_{name_}, level_{level}, range_{range} {}
```

It slightly pains me that there are effectively two string identifiers in this class (`name_` and `name_key_`). I'm guessing the difference between them is effectively that the `core::Identifier` is always lowercase?